sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidRexExecutor.java

```diff
@@ -122,6 +122,7 @@ public void reduce(
             if (exprResult.type() == ExprType.LONG) {
               bigDecimal = BigDecimal.valueOf(exprResult.asLong());
             } else {
+              // if exprResult evaluates to Nan or infinity, this will throw a NumberFormatException
```

Comment:
I wasn't sure how to deal with non numbers here, so I decided to leave the behavior as is and let it fail. It's unclear to me if Calcite will allow us to get here. In my local testing, I've seen NumberFormatExceptions thrown in Calcite when I tried to write sql expressions that would compute to Nan or infinity (like `0D / 0`).

My thoughts were this is an edge case so it's ok to leave this behavior as is.

Reply:
Wondering what is your thought on making this reduce to value of 0 (same as the other stuff)

Reply:
I thought this might produce confusing behavior if the literal was used in a comparison operation. For eg. `1 < POSITIVE_INFINITY` would return false because `POSITIVE_INFINITY` would evaluate to if we changed the code here.

Reply:
I guess the same argument can be made if we compared 2 columns `x < y` and x was `1`, y was `POSITIVE_INFINITY` then the result of the comparison would be false after this change.

Reply:
Do you know if this is mention in Druid docs?

Reply:
I don't think so. I was afraid documenting all the edge cases would make the docs too verbose.